pipeline/frontend/yaml/compiler/compiler.go

```diff
@@ -253,7 +253,7 @@ func (c *Compiler) Compile(conf *yaml_types.Workflow) (*backend_types.Config, er
 		step := c.createProcess(name, container, stepType)
 
 		// inject netrc if it's a trusted repo or a trusted clone-plugin
-		if c.trustedPipeline || (container.IsPlugin() && container.IsTrustedCloneImage()) {
+		if !c.netrcOnlyTrusted || c.trustedPipeline || (c.netrcOnlyTrusted && container.IsPlugin() && container.IsTrustedCloneImage()) {
```

Comment:
Thinking about it I would prefer following cases:
- if trusted repo (needs admin access)
- if plugin and trusted clone image (clone image should not steal /leak the credentials)

New case:
- if `netrcOnlyTrusted` is active only allow plugins as the admin could trust them, so we might need a list of `passNetrcToImages` instead